### PR TITLE
[Samples] Add 17.multilingual-bot sample

### PR DESCRIPTION
**Pull request by Batta32**
Fixes 868

- [X] Migration
- [ ] Testing

## Description

We are adding the sample 17.multilingual-bot in Java having into account the [C# 17.multilingual-bot](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/17.multilingual-bot) present in [BotBuilder-Samples](https://github.com/microsoft/BotBuilder-Samples) repository.

## Specific Changes

  - Generate main structure of the sample
  - Add `bots` folder with the `MultiLingualBot.java` class
  - Add `cards` folder with `welcomeCard.json`
  - Add `deploymentTemplates` folder
  - Add `translation` folder including the `model` folder and classes
  - Add root files
  - Add `application.properties`
  - Add LICENSE document
  - Add README document to describe the steps to run the sample

## Testing

We are currently testing the migrated sample. As soon as it's validated, it will be ready to be merged.

_Current behavior of the Java migrated sample, need more manual testing_
image

### CODE REVIEW

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+// Use the default BotController to receive incoming Channel messages. A custom
+// controller could be used by eliminating this import and creating a new
+// RestController.
+// The default controller is created by the Spring Boot container using
+// dependency injection. The default route is /api/messages.
+@Import({BotController.class})
+
+public class Application extends BotDependencyConfiguration {
+    public static void main(String[] args) {
```
Comment (VictorGrycuk) [inline samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:40]: On my checkstyle, it asks for a javadoc here

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+import com.microsoft.bot.builder.ConversationState;
+import com.microsoft.bot.builder.Storage;
+import com.microsoft.bot.builder.UserState;
+import com.microsoft.bot.integration.AdapterWithErrorHandler;
+import com.microsoft.bot.integration.BotFrameworkHttpAdapter;
+import com.microsoft.bot.integration.Configuration;
+import com.microsoft.bot.integration.spring.BotController;
+import com.microsoft.bot.integration.spring.BotDependencyConfiguration;
+import com.microsoft.bot.sample.multilingual.translation.TranslationMiddleware;
```
Comment (VictorGrycuk) [inline samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:14]: There seem to be a missing import:
`import com.microsoft.bot.sample.multilingual.translation.MicrosoftTranslator;`

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Create the Translation Middleware that will be added to the middleware pipeline in the AdapterWithErrorHandler.
+     * @return TranslationMiddleware
+     */
+    @Bean
+    public TranslationMiddleware getTranslationMiddleware() {
+        Storage storage = this.getStorage();
+        UserState userState = this.getUserState(storage);
+        MicrosoftTranslator microsoftTranslator = this.getMicrosoftTranslator();
```
Comment (VictorGrycuk) [inline samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:79]: Isnt missing a parameter in `this.getMicrosoftTranslator()`?

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import com.microsoft.bot.builder.StatePropertyAccessor;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.CardAction;
+import com.microsoft.bot.schema.ActionTypes;
+import com.microsoft.bot.schema.SuggestedActions;
+import com.microsoft.bot.schema.Attachment;
+import java.io.BufferedReader;
```
Comment (VictorGrycuk) [inline samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java:19]: Add a space between java libs and local libs (com.microsoft)?

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/translation/TranslationMiddleware.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    public CompletableFuture<Void> onTurn(TurnContext turnContext, NextDelegate next) {
+        if (turnContext == null) {
+            throw new IllegalArgumentException("turnContext");
+        }
+
+        return this.shouldTranslate(turnContext).thenCompose(translate -> {
+            if (translate) {
+                if (turnContext.getActivity().getType() == ActivityTypes.MESSAGE) {
+                    return this.translator.translate(turnContext.getActivity().getText(), TranslationSettings.DEFAULT_LANGUAGE).thenApply(text -> {
```
Comment (VictorGrycuk) [inline samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/translation/TranslationMiddleware.java:59]: There are several lines over 120 chars and that fires the checkstyle warning, are we going to follow that rule?

**Review by VictorGrycuk — CHANGES_REQUESTED** [review]
I added comments

File: samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java (hunk 1 above)
Comment (Batta32) [reply samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:40]: Just to confirm, you checkstyle is using the rules of ms?

Comment (VictorGrycuk) [reply samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:41]: Its pointing to the local xml

Comment (Batta32) [reply samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/Application.java:41]: I don't see another Application with comments. Let's wait for feedback of tracy here

**Review by FedericoBernal — APPROVED** [review]

**Pull request closed without merging**